frontend/src/scenes/insights/InsightTabs/InsightDisplayConfig.tsx

```diff
@@ -0,0 +1,177 @@
+import { useValues } from 'kea'
+import { ChartFilter } from 'lib/components/ChartFilter'
+import { CompareFilter } from 'lib/components/CompareFilter/CompareFilter'
+import { DateFilter } from 'lib/components/DateFilter/DateFilter'
+import { IntervalFilter } from 'lib/components/IntervalFilter'
+import { SaveToDashboard } from 'lib/components/SaveToDashboard/SaveToDashboard'
+import { TZIndicator } from 'lib/components/TimezoneAware'
+import { ACTIONS_BAR_CHART_VALUE, ACTIONS_LINE_GRAPH_LINEAR, ACTIONS_PIE_CHART, ACTIONS_TABLE } from 'lib/constants'
+import { featureFlagLogic } from 'lib/logic/featureFlagLogic'
+import React from 'react'
+import { DisplayType, FilterType } from '~/types'
+import { ViewType } from '../insightLogic'
+import { CalendarOutlined } from '@ant-design/icons'
+
+interface InsightDisplayConfigProps {
+    clearAnnotationsToCreate: () => void
+    allFilters: FilterType
+    activeView: ViewType
+    annotationsToCreate: any[] // TODO: Annotate properly
+}
+
+export function InsightDisplayConfig({
+    horizontalUI,
+    ...props
+}: InsightDisplayConfigProps & { horizontalUI: boolean }): JSX.Element {
+    return horizontalUI ? (
+        <HorizontalDefaultInsightDisplayConfig {...props} />
+    ) : (
+        <DefaultInsightDisplayConfig {...props} />
+    )
+}
+
+const showIntervalFilter = function (activeView: ViewType, filter: FilterType): boolean {
+    switch (activeView) {
+        case ViewType.FUNNELS:
+            return filter.display === ACTIONS_LINE_GRAPH_LINEAR
+        case ViewType.RETENTION:
+        case ViewType.PATHS:
+            return false
+        case ViewType.TRENDS:
+        case ViewType.STICKINESS:
+        case ViewType.LIFECYCLE:
+        case ViewType.SESSIONS:
+        default:
+            return ![ACTIONS_PIE_CHART, ACTIONS_TABLE, ACTIONS_BAR_CHART_VALUE].includes(filter.display || '') // sometimes insights aren't set for trends
+    }
+}
+
+const showChartFilter = function (activeView: ViewType, featureFlags: Record<string, boolean>): boolean {
+    switch (activeView) {
+        case ViewType.TRENDS:
+        case ViewType.STICKINESS:
+        case ViewType.SESSIONS:
+        case ViewType.RETENTION:
+            return true
+        case ViewType.FUNNELS:
+            return featureFlags['funnel-trends-1269']
+        case ViewType.LIFECYCLE:
+        case ViewType.PATHS:
+            return false
+        default:
+            return true // sometimes insights aren't set for trends
+    }
+}
+
+const showDateFilter = {
```

Comment:
Just wondering -- why are some of these flag getters objects, and some others are functions? Should we prefer one approach or the other? (I personally prefer the functions with `switch`)

Reply:
Hmmm that's a good question, not sure actually (it's code quite old), but I guess that we have the functions for logic that requires checking other variables and just plain constants for when it's just a static definition. Will keep in mind for a refactoring.